packages/app/src/specs/RunStatusDots.vue

```diff
@@ -0,0 +1,206 @@
+import RunningIcon from '~icons/cy/running-outline_x16.svg'
+import SpecRunSummary from './SpecRunSummary.vue'
+import { gql } from '@urql/vue'
+
+gql`
+fragment RunStatusDots on RemoteFetchableCloudProjectSpecResult {
+  id
+  data {
+    ... on CloudProjectSpecNotFound {
+      retrievedAt
+    }
+    ... on CloudProjectSpec {
+      id
+      retrievedAt
+      specRuns(first: 4, fromBranch: $fromBranch) {
+        nodes {
+          id
+          runNumber
+          testsFailed{
+            min
+            max
+          }
+          testsPassed{
+            min
+            max
+          }
+          testsPending{
+            min
+            max
+          }
+          testsSkipped{
+            min
+            max
+          }
+          createdAt
+          groupCount
+          specDuration{
+            min
+            max
+          }
+          status
+          url
+        }
+      }
+    }
+  }
+}
+`
+
+const props = defineProps<{
+  gql: RunStatusDotsFragment | null
+  specFileName: string
+  specFileExtension: string
+}>()
+
+const runs = computed(() => {
+  return props.gql?.data?.__typename === 'CloudProjectSpec' ? props.gql?.data?.specRuns?.nodes ?? [] : []
+})
+
+const dotClasses = computed(() => {
+  const statuses = ['placeholder', 'placeholder', 'placeholder']
+
+  if (runs.value && runs.value.length > 0) {
+    // skip the 0th index as it represents the latest run and dotClasses
+    // is meant to represent the 3 prior runs.
+    for (let i = 1; i < Math.min(runs.value.length, 4); i++) {
+      statuses[i - 1] = runs.value[i]?.status ?? ''
+    }
+  }
+
+  return statuses.reverse().map((s) => {
+    switch (s) {
+      case 'PASSED':
+        return 'icon-light-jade-400'
+      case 'RUNNING':
+        return 'icon-light-indigo-400'
+      case 'FAILED':
+        return 'icon-light-red-400'
+      case 'ERRORED':
+      case 'OVERLIMIT':
+      case 'TIMEDOUT':
+        return 'icon-light-orange-400'
+      case 'NOTESTS':
+        return 'icon-light-gray-400'
+      case 'CANCELLED':
+      default:
+        return 'icon-light-gray-300'
+    }
+  })
+})
+
+const latestRun = computed(() => {
+  if (runs.value == null || runs.value.length === 0 || runs.value[0] === null) {
+    return null
+  }
+
+  return runs.value[0]
+})
+
+const latestStatus = computed(() => {
+  const run = latestRun.value
+
+  if (!run) {
+    return { icon: PlaceholderIcon, spin: false }
+  }
+
+  switch (run.status) {
+    case 'PASSED':
+      return { icon: PassedIcon, spin: false }
+    case 'RUNNING':
+      return { icon: RunningIcon, spin: true }
+    case 'FAILED':
+      return { icon: FailedIcon, spin: false }
+    case 'ERRORED':
+    case 'OVERLIMIT':
+    case 'TIMEDOUT':
+      return { icon: ErroredIcon, spin: false }
+    case 'NOTESTS':
+    case 'CANCELLED':
+      return { icon: CancelledIcon, spin: false }
+    default:
+      return { icon: PlaceholderIcon, spin: false }
+  }
+})
```

Comment:
Can bypass the return on line 175 by using optional chaining. Up to you, just personal preference

```suggestion
  switch (latestRun.value?.status) {
    case 'PASSED':
      return { icon: PassedIcon, spin: false }
    case 'RUNNING':
      return { icon: RunningIcon, spin: true }
    case 'FAILED':
      return { icon: FailedIcon, spin: false }
    case 'ERRORED':
    case 'OVERLIMIT':
    case 'TIMEDOUT':
      return { icon: ErroredIcon, spin: false }
    case 'NOTESTS':
    case 'CANCELLED':
      return { icon: CancelledIcon, spin: false }
    default:
      return { icon: PlaceholderIcon, spin: false }
  }
})
```